### PR TITLE
CRM-18286 - Use localStorage to remember datatable page length

### PR DESCRIPTION
**Pull request by colemanw**
https://issues.civicrm.org/jira/browse/CRM-18286

### CODE REVIEW

File: js/Common.js [hunk 1]
```diff
@@ -728,8 +728,20 @@ if (!CRM.vars) CRM.vars = {};
   };
 
   $.fn.crmAjaxTable = function() {
+    // Strip the ids from ajax urls to make pageLength storage more generic
+    function simplifyUrl(ajax) {
+      // Datatables ajax prop could be a url string or an object containing the url
+      var url = typeof ajax === 'object' ? ajax.url : ajax;
```
Comment (AkA84) [inline js/Common.js]: I think this could be simplified as:
```js
var url = ajax.url || ajax
```

**Comment by AkA84** [comment]
Successfully replicated the issue on the `master` branch, on the contact's Activities page and on the Manag Group pages.

On the PR's branch, the issue is gone, the datatable page length being remembered after page refresh on all the major browsers

File: js/Common.js [hunk 2]
```diff
@@ -728,8 +728,20 @@ if (!CRM.vars) CRM.vars = {};
...
+      return typeof url === 'string' ? url.replace(/[&?]\w*id=\d+/g, '') : null;
```
Comment (AkA84) [inline js/Common.js]: What if you remove the querystring altogether? Is that too generic?

Comment (colemanw) [inline js/Common.js]: Yes, it would apply to all datatables (at least in WP and Joomla where the civi path is part of the query string).

**Comment by totten** [comment]
There was additional feedback from the 4.7.8 spreadsheet, "The patch works as intended, left a couple of suggestions for improving the solution".

If @colemanw and @AkA84 both say it works, then I'm inclined to merge. :) 

Will leave it to @colemanw whether to submit follow-up with improvements.

**Pull request merged**